nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -145,7 +159,15 @@ def get_generations(profile: Optional[str] = None) -> List[Tuple[Optional[str],
     gen_lines.pop()
 
     configurationLimit = @configurationLimit@
-    return [ (profile, int(line.split()[0])) for line in gen_lines ][-configurationLimit:]
+    return [ (profile, int(line.split()[0]), None) for line in gen_lines ][-configurationLimit:]
+
+
+def get_specialisations(profile, generation, _):
```

Comment:
I just guessed types here. Please fill in the correct ones:

```suggestion
def get_specialisations(profile: str, generation: int, _) -> List[Tuple[str, int, ?]]:
```

Reply:
`gen` is a `Tuple[Optional[str], int]`, where does third variable comes from?

Reply:
I think you were looking at a stale diff, I've already typed these.